### PR TITLE
Feature add stripe to react component

### PR DESCRIPTION
**Pull request by OleksaRiabukha**
Will add stripe checkout for coupon buying and call to coupon API for the coupon generation upon successful payment.

### CODE REVIEW

File: app/controllers/concerns/stripe_checkout.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module StripeCheckout
-  def self.create_stripe_checkout(cart, success_url, cancel_url, current_user)
-    amount = (cart.total_cart_price * 100).round
+  def self.create_stripe_checkout(amount, payment_for, success_url, cancel_url, current_user, model)
+    amount = (amount.to_f.round(2) * 100).to_i
```
Comment (BohdanBusko) [inline app/controllers/concerns/stripe_checkout.rb]: to_d we don't use float we use decimal

File: app/javascript/bundles/Coupons/components/Coupons.jsx [hunk 2]
```diff
@@ -13,11 +22,37 @@ const Coupons = () => {
     criteriaMode: 'all',
   });
 
-  const onSubmit = (data) => console.log(data);
+  const csfrHeader = {
+    headers: {
+      'X-CSRF-Token': ReactOnRails.authenticityToken(),
+    },
+  };
+
```
Comment (BohdanBusko) [inline app/javascript/bundles/Coupons/components/Coupons.jsx:30]: remove this enter

File: app/javascript/bundles/Coupons/components/Coupons.jsx [hunk 3]
```diff
@@ -1,9 +1,18 @@
-import React from 'react';
-import { ErrorMessage } from '@hookform/error-message';
+import React, {useState} from 'react';
+import ReactOnRails from 'react-on-rails';
+import {useForm} from 'react-hook-form';
```
Comment (BohdanBusko) [inline app/javascript/bundles/Coupons/components/Coupons.jsx]: leave spaces

File: app/javascript/bundles/Coupons/components/Coupons.jsx [hunk 4]
```diff
@@ -13,11 +22,37 @@ const Coupons = () => {
...
 
-  const onSubmit = (data) => console.log(data);
+  const csfrHeader = {
+    headers: {
+      'X-CSRF-Token': ReactOnRails.authenticityToken(),
+    },
+  };
+
+
+  const addCoupontoUser = async (data) => {
```
Comment (BohdanBusko) [inline app/javascript/bundles/Coupons/components/Coupons.jsx]: addCouponToUser...

File: app/javascript/bundles/Coupons/components/Coupons.jsx [hunk 5]
```diff
@@ -13,11 +22,37 @@ const Coupons = () => {
...
+        amount: data.amount,
+        for_present: data.forPresent,
+      },
+    };
+
+    await axios
+        .post(props.path, requestBody, csfrHeader)
+        .then(function(response) {
+          if (response.status === 200) {
+            window.location.href = response.data.session.url;
```
Comment (BohdanBusko) [inline app/javascript/bundles/Coupons/components/Coupons.jsx:43]: do we really need this redirect?

File: app/javascript/bundles/Coupons/components/ErrorModal.jsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import {Button, Modal} from 'react-bootstrap';
```
Comment (BohdanBusko) [inline app/javascript/bundles/Coupons/components/ErrorModal.jsx]: spaces

File: app/services/coupon_creator.rb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+                            title: #{errors['title']},
+                            details: #{errors['details']}).squish
+
+      logger.error message
+    end
+
+    private
+
+    def connect
+      Faraday.new(
```
Comment (BohdanBusko) [inline app/services/coupon_creator.rb:35]:
```suggestion
      Faraday.new(
        url: URL,
        headers: { 
          'Content-Type' => 'application/json', 
          'Authorization' => API_KEY 
        }
      )
```

File: app/controllers/orders_controller.rb [hunk 8]
```diff
@@ -35,7 +37,7 @@ def create
     end
   end
 
-  def successful_checkout
+  def successful_order_checkout
```
Comment (BohdanBusko) [inline app/controllers/orders_controller.rb:40]: we don't really need such actions you can name it successful_order_checkout_url and return restaurant_path also move it to private section

File: app/javascript/bundles/Coupons/components/Coupons.jsx (hunk 5 above)
Comment (OleksaRiabukha) [reply app/javascript/bundles/Coupons/components/Coupons.jsx:43]: React won't redirect to Stripe checkout without it

File: app/controllers/orders_controller.rb (hunk 8 above)
Comment (OleksaRiabukha) [reply app/controllers/orders_controller.rb:40]: But how the router will find this method in such a case? I mean, Stripe Session accepts URL as a parameter and redirects user upon successful session completion. If it won't be an action, how can we redirect the user?

**Review by BohdanBusko — APPROVED** [review]

**Pull request merged**